node/src/actors/chain_manager/mod.rs

```diff
@@ -829,6 +834,56 @@ impl ChainManager {
         }
     }
 
+    fn update_bit_counter(&mut self, v: u32, epoch: u32) {
+        for n in 0..32i8 {
+            if let Some(mut bit_counter) = self.chain_state.bit_tapi_counter.get_mut(&n, &epoch) {
+                if !self
+                    .chain_state
+                    .wip_activation
+                    .contains_key(&bit_counter.wip)
+                    && is_bit_n_activated(v, n)
+                {
+                    bit_counter.votes += 1;
+
+                    if (epoch + bit_counter.init) % bit_counter.period == 0 {
+                        if (bit_counter.votes * 100) / bit_counter.period >= 80 {
+                            // An offset of 21 is added to ensure that the activation of the WIP is
+                            // achieved with consolidated blocks
+                            self.chain_state
+                                .wip_activation
+                                .insert(bit_counter.wip.clone(), epoch + 21);
+                        }
+                        bit_counter.votes = 0;
+                    }
+                }
+            }
+        }
+    }
+
+    #[allow(clippy::cast_possible_truncation, clippy::cast_possible_wrap)]
+    fn update_wip_information(&mut self) {
```

Comment:
Is this method used to initialize the `BitTapiCounter` once or is this something that should be called periodically to update the structure?

```suggestion
    fn initialize_wip_information(&mut self) {
```